catalyst_voices_packages/catalyst_cardano_serialization/lib/src/transaction.dart

```diff
@@ -66,38 +66,73 @@ final class Transaction extends Equatable implements CborEncodable {
 ///
 /// Does not contain the witnesses which are used to verify the transaction.
 final class TransactionBody extends Equatable implements CborEncodable {
-  /// The transaction inputs.
+  /// The transaction inputs. tag: 0
   final Set<TransactionInput> inputs;
 
-  /// The transaction outputs.
-  final List<TransactionOutput> outputs;
+  /// The transaction outputs. tag: 1
+  final List<ShelleyMultiAssetTransactionOutput> outputs;
 
-  /// The fee for the transaction.
+  /// The fee for the transaction. tag: 2
   final Coin fee;
 
-  /// The absolute slot value before the tx becomes invalid.
+  // > Note: All properties below are optional.
+  /// The absolute slot value before the tx becomes invalid. tag: 3
   final SlotBigNum? ttl;
 
+  /// Certificates in an ordered set. tag: 4
+  /// Withdrawals map of stake address, coin. tag: 5
+
   /// The hash of the optional [AuxiliaryData]
-  /// which is the metadata of the transaction.
+  /// which is the metadata of the transaction. tag: 7
   final AuxiliaryDataHash? auxiliaryDataHash;
 
+  /// Validity interval start as integer. tag: 8
+  final SlotBigNum? validityStart;
+
+  /// Mint as a non-zero uint64 multiasset. tag: 9
+  final MultiAsset? mint;
+
+  /// Tag 10 is unimplemented.
+  /// Script data hash28. tag: 11
+  final ScriptDataHash? scriptDataHash;
+
+  /// Tag 12 is unimplemented.
+  /// Collateral inputs as nonempty set. tag: 13
+  final Set<TransactionInput>? collateralInputs;
+
   /// The list of public key hashes of addresses
   /// that are required to sign the transaction.
+  /// Nonempty set of addr keyhash. tag: 14
   final Set<Ed25519PublicKeyHash>? requiredSigners;
 
-  /// Specifies on which network the code will run.
+  /// Specifies on which network the code will run. Network ID 0/1. tag: 15
   final NetworkId? networkId;
 
+  /// Collateral return's transaction output. tag: 16
+  final ShelleyMultiAssetTransactionOutput? collateralReturn;
+
+  /// Total collateral as coin (uint64). tag: 17
+  final Coin? totalCollateral;
+
+  /// Reference inputs as nonempty set of transaction inputs. tag: 18
+  final Set<TransactionInput>? referenceInputs;
+
   /// The default constructor for [TransactionBody].
   const TransactionBody({
-    required this.inputs,
-    required this.outputs,
-    required this.fee,
-    this.ttl,
-    this.auxiliaryDataHash,
-    this.requiredSigners,
-    this.networkId,
+    required this.inputs, // tag: 0
```

Comment:
Would it be sufficient to only document the tag inside the field comment and remove it from the constructor? The point is that `// tag: 0` is not going to appear in the contextual menu in the IDE when you hover over the constructor.

To know the tag value you need to invoke the contextual help for the field itself which means the tag here is not needed that much.